src/app/components/query-explainer/query-explainer.types.ts

```diff
@@ -0,0 +1,41 @@
+export interface QueryExplainerResponse {
+    input: {
+        query: string;
+        queryDialect: string;
+        dataFormat: string;
+        schemaFormat: string;
+        include: string[];
+        skip: number;
+        limit: number;
+        datasets: [{ alias: string; blockHash: string; id: string }];
+    };
+    commitment: {
+        inputHash: string;
+        outputHash: string;
+        subQueriesHash: string;
+    };
+    proof: {
+        proofValue: string;
+        type: string;
+        verificationMethod: string;
+    };
+    output?: {
+        data: [string[]];
+        dataFormat: string;
+        schema: {
+            fields: [{ name: string }];
+        };
+    };
+}
+
+export interface VerifyQueryResponse {
+    ok: boolean;
+    error?: {
+        kind: string;
+        message: string;
+        actual_hash?: string;
+        expected_hash?: string;
+        dataset_id?: string;
+        block_hash?: string;
+    };
```

Comment:
As I understand, this is modeling a few types of errors which may nor may not have some of these parameters.
I prefer if we define a separate clean structure per each type of error that is hidden here, and union them in the field. The code that is using this will first test for `kind`, which is shared, and downcast to particular good error structures depending on the kind.

Reply:
Added separate error interface for all kind of errors